core/agents/architect.py

```diff
@@ -4,6 +4,7 @@
 
 from core.agents.base import BaseAgent
 from core.agents.convo import AgentConvo
+from core.agents.mixins import SystemDependencyCheckerMixin
```

Comment:
In my opinion, we shouldn't use this as mixin. We should always go though whole flow and each agent should do it's own job.